pasqal_cloud/batch.py

```diff
@@ -143,7 +159,20 @@ def declare_complete(
                     self.id,
                 )
             for job_rsp in batch_rsp["jobs"]:
-                self.jobs[job_rsp["id"]] = Job(**job_rsp)
+                job = Job(**job_rsp)
```

Comment:
As ordered jobs is a list of dicts, we need a list comprehension list to be able to get the index of the batch we want to update, and then replace it by its updated data.

Reply:
Updated, please let me know if it looks better, my first try was not that much readable, I think this one is better. However there might be a potential error of IndexError if there is an additional job in the job_rsp. Idk if this case if possible though